core/base/src/TSystem.cxx

```diff
@@ -3379,7 +3379,7 @@ int TSystem::CompileMacro(const char *filename, Option_t *opt,
       // the ACLiC library.
       if (useCxxModules && !produceRootmap) {
          std::function<bool(const char *)> LoadLibF = [](const char *dep) {
-            return gInterpreter->Load(dep, /*system*/ false) >= 0;
+            return gSystem->Load(dep) >= 0;
```

Comment:
This is likely a step in the right direction but more needs to be done as ```gSystem->Load``` is implemented in term of ```gInterpreter->Load```.  So it is really the behavior in gInterpreter->Load that have been (very backwardly incompatible) altered (it nows reload shared libraries and re-execute their static initialization),

Reply:
Looks like whenever we want to permanently load (and not reload if already loaded) the library we need to set the `system` optional parameter to `true`: https://github.com/root-project/root/blob/master/core/metacling/src/TCling.cxx#L3355